meson.build

```diff
@@ -636,34 +636,40 @@ endif
 
 ############################################################
 
-gperf = find_program('gperf')
-
-gperf_test_format = '''
-#include <string.h>
-const char * in_word_set(const char *, @0@);
-@1@
-'''
-gperf_snippet_format = 'echo foo,bar | @0@ -L ANSI-C'
-gperf_snippet = run_command('sh', '-c', gperf_snippet_format.format(gperf.path()))
-gperf_test = gperf_test_format.format('size_t', gperf_snippet.stdout())
-if cc.compiles(gperf_test)
-        gperf_len_type = 'size_t'
-else
-        gperf_test = gperf_test_format.format('unsigned', gperf_snippet.stdout())
+gperf = find_program('gperf', required: false)
+if not gperf.found()
+        gperf = disabler()
```

Comment:
> This still looks wrong. If shouldn't be necessary to redefine the gperf object, the thing returned from find_program should b enough.

I did that to support meson versions below `0.49.0`.
When using `0.49.0` and above you can just do this:
```python
gperf = find_program('gperf', required: false, disabler: true)
```

> I'm also not convinced that we need the separate warning... find_program already says "NO" in red. It is not very visible, but can be considered enough.

I added the warning there based off of the previous error text emitted by `find_program`, I figured for these dependencies it made sense to do so to increase visibility since a large amount of systemd features do depend on them. I can remove the warning if you think it's not needed.